From 754e4319b38081d49a52b61c7df4a92886ce01b6 Mon Sep 17 00:00:00 2001
From: Andrew Wang <wang@apache.org>
Date: Mon, 7 Mar 2016 14:00:02 -0800
Subject: [PATCH 1310/2848] HADOOP-12901. Add warning log when
 KMSClientProvider cannot create a connection to
 the KMS server. (Xiao Chen via wang)

(cherry picked from commit 391da36d93358038c50c15d91543f6c765fa0471)
(cherry picked from commit fed2339e285e84a40514f877703b2712a2305a6f)
(cherry picked from commit 352d5a7e88eff165274646ef2d79d385fcf42b70)

Change-Id: I8afef82023f700ee33bcc0014b09a30dc3f2f38e
---
 .../hadoop/crypto/key/kms/KMSClientProvider.java   |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java
index 8e6e220..19b6b19 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/KMSClientProvider.java
@@ -41,6 +41,8 @@
 import org.apache.hadoop.util.HttpExceptionUtils;
 import org.apache.http.client.utils.URIBuilder;
 import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.net.ssl.HttpsURLConnection;
 
@@ -84,6 +86,9 @@
 public class KMSClientProvider extends KeyProvider implements CryptoExtension,
     KeyProviderDelegationTokenExtension.DelegationTokenExtension {
 
+  private static final Logger LOG =
+      LoggerFactory.getLogger(KMSClientProvider.class);
+
   private static final String INVALID_SIGNATURE = "Invalid signature";
 
   private static final String ANONYMOUS_REQUESTS_DISALLOWED = "Anonymous requests are disallowed";
@@ -486,6 +491,9 @@ public HttpURLConnection run() throws Exception {
         }
       });
     } catch (IOException ex) {
+      if (ex instanceof SocketTimeoutException) {
+        LOG.warn("Failed to connect to {}:{}", url.getHost(), url.getPort());
+      }
       throw ex;
     } catch (UndeclaredThrowableException ex) {
       throw new IOException(ex.getUndeclaredThrowable());
-- 
1.7.9.5

